### PR TITLE
elmPackages.elm-format: disable checks

### PR DESCRIPTION
**Pull request by ncfavier**
Fixes the build with optparse-applicative 0.17

### CODE REVIEW

**Review by cdepillabout — APPROVED** [review]
LGTM, thanks!

File: pkgs/development/compilers/elm/default.nix [hunk 1]
```diff
@@ -37,6 +37,7 @@ let
       */
       elm-format = justStaticExecutables (overrideCabal (drv: {
         jailbreak = true;
+        doCheck = assert (drv.version == "0.8.5"); false; # golden tests fail with optparse-applicative 0.17
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/default.nix:40]: is it safe to just skip the golden test? Elm format producing different result than expected would not be desirable.

Comment (avh4) [reply pkgs/development/compilers/elm/default.nix:40]: As noted here https://github.com/NixOS/nixpkgs/pull/202022#issuecomment-1343798674 currently the failure is in the usage help due to the jailbreak of optparse-applicative.

Given the goal of fixing `haskell-updates` with limited volunteer time by Dec 22, I think this quick fix is reasonable.  I guess the risk is that some other future jailbreak could break something more critical 🤷.  I should have elm-format 0.8.6 released in Jan or Feb, which will upgrade optparse-applicative and fix the tests upstream.

That said, anyone wants to put in the work to do a different fix here, I'd support that too :)

**Comment by avh4** [comment]
Thanks!

File: pkgs/development/compilers/elm/default.nix (hunk 1 above)
Comment (ncfavier) [reply pkgs/development/compilers/elm/default.nix:40]: (that's why I added `assert (drv.version == "0.8.5")`: the tests are only known to be safely ignorable on this version)

**Pull request merged**